### PR TITLE
refactor contact models

### PR DESCRIPTION
**Pull request by yenda**
- move models.contact and models.contacts to contact.core
- add tags field to contact and chat

### Summary:

This PR cleans up the different contact namespaces and adds a tags fields to chats and contacts in preparation for future PR that will allow user to filter chats based on those tags

#### Platforms (optional)
- Android
- iOS
- macOS
- Linux

#### Areas that maybe impacted
**Functional**
- 1-1 chats
- public chats
- group chats

### Steps to test:
- Test the various add-contact flows
- Test migration from current version to this one

status: ready

### CODE REVIEW

File: src/status_im/contact/core.cljs [hunk 1]
```diff
@@ -48,6 +67,16 @@
               (add-new-contact contact)
               (send-contact-request contact))))
 
+(fx/defn add-contact-tag [{:keys [db] :as cofx} whisper-id tag]
```
Comment (cammellos) [inline src/status_im/contact/core.cljs]: Are these fns used or it's something you will be adding at a later stage?

**Review by cammellos — APPROVED** [review]

File: src/status_im/contact/core.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/contact/core.cljs]: it is something that will be used at a later stage, it's fx producing functions to add/remove a tag on a contact

**Review by dmitryn — APPROVED** [review]

File: src/status_im/data_store/realm/schemas/account/contact.cljs [hunk 2]
```diff
@@ -25,3 +25,30 @@
...
+(def v2 {:name       :contact
+         :primaryKey :whisper-identity
+         :properties {:address          {:type :string :optional true}
+                      :whisper-identity :string
+                      :name             {:type :string :optional true}
+                      :photo-path       {:type :string :optional true}
+                      :last-updated     {:type :int :default 0}
+                      :last-online      {:type :int :default 0}
+                      :pending?         {:type :bool :default false}
+                      :hide-contact?    {:type :bool :default false}
```
Comment (janherich) [inline src/status_im/data_store/realm/schemas/account/contact.cljs:38]: I'm not 100% sure, but this is probably not needed anymore and was relevant when we had "bot" contacts, seems like a good opportunity to get rid of it.

File: src/status_im/data_store/realm/schemas/account/contact.cljs [hunk 3]
```diff
@@ -25,3 +25,30 @@
...
+                      :status           {:type :string :optional true}
+                      :fcm-token        {:type :string :optional true}
+                      :description      {:type :string :optional true}
+                      :public-key       {:type     :string
+                                         :optional true}
+                      :dapp?            {:type    :bool
+                                         :default false}
+                      :dapp-url         {:type     :string
+                                         :optional true}
+                      :bot-url          {:type     :string
```
Comment (janherich) [inline src/status_im/data_store/realm/schemas/account/contact.cljs:48]: Same as above.

**Review by janherich — APPROVED** [review]

File: src/status_im/data_store/realm/schemas/account/contact.cljs (hunk 3 above)
Comment (yenda) [reply src/status_im/data_store/realm/schemas/account/contact.cljs:48]: so what are the fields that can be deleted ? bot-url and hide-contact? ? Is there any other ?

**Comment by lukaszfryc** [comment]
@yenda should I wait for more commits?

**Comment by yenda** [comment]
@lukaszfryc  no

**Pull request merged**